### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by krasimirkostadinow**

### CODE REVIEW

File: clientFactory/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    config: {},
+    context: {}
+};
+
+module.exports = (clientConfig) => {
+    let config = Object.assign({}, defaultConfig, clientConfig);
+    config.type = config.options.watch ? `${config.type}Watch` : config.type;
+    let Constructor = clients[config.type];
+    if (!Constructor) {
+        throw new Error(`Uknown registry type: ${config.type}! Available types: ${Object.keys(clients).join(', ')}`);
```
Comment (stambata) [inline clientFactory/index.js]: Hardcode available types to 'consul', 'ut'

File: clientFactory/consul/utils.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                })) {
+                    all.push(getServiceDefinition(record, raw));
+                }
+                return all;
+            }, []);
+        }
+        return records.map((record) => getServiceDefinition(record, raw));
+    },
+    compare: function(service, localData, remoteData) {
+        return Promise.resolve()
```
Comment (stambata) [inline clientFactory/consul/utils.js]: return boolean directly

File: clientFactory/consul/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                promisify: true
+            }
+        ));
+        return this.consul.agent.self();
+    }
+
+    serviceAdd(definition) {
+        var msg = utils.encode(definition, this.context);
+        return this.consul.agent.service.register(msg)
+            .then(() => Promise.resolve({success: true}));
```
Comment (stambata) [inline clientFactory/consul/index.js]: return {success: true} directly.

File: clientFactory/utWatch/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                    });
+            })
+            .then((records) => {
+                criteria.raw = false;
+                return consulUtils.decode(records, criteria);
+            });
+    }
+
+    initWs(host, service) {
+        let socketHost = host || this.config.url.match(/\/+(.*)/)[1];
```
Comment (stambata) [inline clientFactory/utWatch/index.js]: check if this will work when there's no '/' at the end

File: clientFactory/utWatch/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+        this.ws[service] = new WebSocket(`ws://${socketHost}/serviceRegistry/${service}`);
+        this.ws[service].on('message', (data) => {
+            let msg = JSON.parse(data);
```
Comment (stambata) [inline clientFactory/utWatch/index.js]: surround try catch

File: index.js [hunk 6]
```diff
@@ -27,6 +22,13 @@ class RegistryPort extends ScriptPort {
                     .then(() => super[method]());
             };
         });
+
+        client.on('change', (data) => {
+            if (client.options.watchMethod !== undefined) {
+                return this.bus.importMethod(client.options.watchMethod)(data);
```
Comment (stambata) [inline index.js]: add catch an this.log && this.log.error(error)

File: readme.md [hunk 7]
```diff
@@ -1,3 +1,140 @@
...
+
+## Usage
+
+The registry port can be initialized in several different ways depending on the implementation requirements. Simple configuration can be found bellow:
+
+```json
+    "registry": {
+        "client": {
+            "type": "consul",
+            "host": "127.0.0.1",
```
Comment (stambata) [inline readme.md]: add host and port in config

File: clientFactory/index.js (hunk 1 above)
Comment (krasimirkostadinow) [reply clientFactory/index.js]: done

**Pull request merged**